### PR TITLE
[PCX-3502] Validate checkbox states

### PR DESCRIPTION
**Pull request by Sorix**
> https://optile.atlassian.net/browse/PCX-3502

## Overview

Front-end validation for switches with required modes. If one or more of these switches are not ticked, the `requiredMessage` appears underneath them and the form is not submitted.

## Implementation details

1. [Create `ExtraElementCheckbox` model with validation logic which conforms `Validatable` protocol](https://github.com/optile/checkout-ios/pull/231/commits/f1d832912dc75f23744f7dd2c1c87ab3e9b2bfc6)
2. [Perform validation on checkbox’s value change](https://github.com/optile/checkout-ios/pull/231/commits/fae8ad925c95934fa1949cee545edbef4927bc66)
3. [Display validation errors in UI](https://github.com/optile/checkout-ios/pull/231/commits/036335294933d202a0c9fe960112e2b5ff90b36c)
4. [Scroll to the top error on the final validation](https://github.com/optile/checkout-ios/pull/231/commits/34599d3a7eeb561d8c797d500c4af42b0fbcc6d5)

## Checklist

- ~Unit tests~
- [x] UI tests
- [x] QA

## Demo

| Before | After |
| - | - |
| <IMG1> | <IMG2> |

| Success | Failure |
| - | - |
| <IMG1> | <IMG2> |

### CODE REVIEW

**Comment by Sorix** [comment]
@caiomello could you add UI logic in these methods?
https://github.com/optile/checkout-ios/blob/dea3aa92905a8576f199cf7cb91d07a5b99b39a2/Sources/PayoneerCheckout/UI/Scenes/Input/Table/Cell/Input.Table.CheckboxViewCell.swift#L134-L140

**Review by izpakla — APPROVED** [review]

File: ExampleCheckout/UITests/ExtraElementsCheckboxTests.swift [hunk 1]
```diff
@@ -72,4 +72,53 @@ final class ExtraElementsCheckboxTests: NetworksTests {
             XCTAssertEqual(checkbox.value as! String, "1")
         }
     }
+
+    // MARK: - Validation
+
+    /// Test validation error message appears for switches with `REQUIRED` mode when switch is changed.
+    func testValidationMessagesOnSwitchChange() throws {
```
Comment (caiomello) [inline ExampleCheckout/UITests/ExtraElementsCheckboxTests.swift:79]: Just 2 tests? Do they cover everything?

File: Sources/PayoneerCheckout/UI/Scenes/Input/Model/Field/Checkbox/Input.Field.ExtraElementCheckbox.swift [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+        var validationErrorText: String?
+
+        // MARK: Initializer
+
+        init(extraElementName: String, isOn: Bool, label: NSAttributedString, isRequired: Requirement) {
+            self.isRequired = isRequired
+
+            super.init(id: .extraElement(extraElementName), isOn: isOn, label: label)
+        }
```
Comment (caiomello) [inline Sources/PayoneerCheckout/UI/Scenes/Input/Model/Field/Checkbox/Input.Field.ExtraElementCheckbox.swift:26]:
```suggestion
        init(extraElementName: String, isOn: Bool, label: NSAttributedString, isRequired: Requirement) {
            self.isRequired = isRequired
            super.init(id: .extraElement(extraElementName), isOn: isOn, label: label)
        }
```

File: Sources/PayoneerCheckout/UI/Scenes/Input/Model/Field/Checkbox/Input.Field.ExtraElementCheckbox.swift [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            if #available(iOS 14.0, *) {
+                logger.error("ExtraElementCheckbox is not required but got validation error, programmatic error. Name=\(self.id.textValue)")
+            }
+            return String()
+        case .required(let requiredMessage):
+            return requiredMessage
+        }
+    }
+
+    var validationRule: Input.Field.Validation.Rule? { return nil }
```
Comment (caiomello) [inline Sources/PayoneerCheckout/UI/Scenes/Input/Model/Field/Checkbox/Input.Field.ExtraElementCheckbox.swift:50]:
```suggestion
    var validationRule: Input.Field.Validation.Rule? {
        return nil
    }
```

File: Sources/PayoneerCheckout/UI/Scenes/Input/Table/Cell/Input.Table.CheckboxViewCell.swift [hunk 4]
```diff
@@ -80,7 +42,62 @@ extension Input.Table {
     }
 }
 
-// MARK: - Cell configuration
+// MARK: - Layout & initial configuration
+
+private extension Input.Table.CheckboxViewCell {
+    func configureTextView() {
+        textView.textColor = CheckoutAppearance.shared.primaryTextColor
```
Comment (caiomello) [inline Sources/PayoneerCheckout/UI/Scenes/Input/Table/Cell/Input.Table.CheckboxViewCell.swift:49]: Is a `UITextView` needed? Why not a label?

**Review by caiomello — APPROVED** [review]

File: Sources/PayoneerCheckout/UI/Scenes/Input/Table/Cell/Input.Table.CheckboxViewCell.swift (hunk 4 above)
Comment (Sorix) [reply Sources/PayoneerCheckout/UI/Scenes/Input/Table/Cell/Input.Table.CheckboxViewCell.swift:49]: We use url links inside text views.

File: ExampleCheckout/UITests/ExtraElementsCheckboxTests.swift (hunk 1 above)
Comment (Sorix) [reply ExampleCheckout/UITests/ExtraElementsCheckboxTests.swift:79]: I think yes. It checks existence or non-existence validation messages when you click on a switch or "Pay" button (4 total cases in total).

**Review by izpakla — APPROVED** [review]

**Review by caiomello — APPROVED** [review]

**Pull request merged**